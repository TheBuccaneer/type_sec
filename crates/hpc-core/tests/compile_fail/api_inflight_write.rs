use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();

    // Buffer erzeugen und beschreiben → Ready
    let buf = ctx.create_buffer::<u8>(16).unwrap()
        .enqueue_write(&queue, &[0u8; 16]).unwrap();

    // Kernel starten → Result wird InFlight
    let (inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();

    // Das hier darf nicht gehen: Write auf InFlight
    inflight.overwrite_blocking(&queue, &[1u8; 16]).unwrap();
}