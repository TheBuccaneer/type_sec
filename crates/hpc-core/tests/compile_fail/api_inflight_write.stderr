error[E0599]: no method named `write_blocking` found for struct `DeviceBuffer<'_, u8, InFlight>` in the current scope
  --> tests/compile_fail/api_inflight_write.rs:20:14
   |
20 |     inflight.write_blocking(&queue, &mut out).unwrap();
   |              ^^^^^^^^^^^^^^ method not found in `DeviceBuffer<'_, u8, InFlight>`
   |
   = note: the method was found for
           - `DeviceBuffer<'brand, T, Mapped>`
           - `DeviceBuffer<'brand, T, Written>`
